### DISCUSSION TITLE
How to handle schema types that have the same names but different case.

### DISCUSSION THREAD

**Discussion by devdamone** [Netflix/dgs-framework#1478] [Q&A] [answered]
I am new to using DGS (and even graphql in general), but so far this seems like a great project and I do like the code generation over some other libraries I have tried.

Background:
When using dgs-codegen, it seems to create a `DgsConstants` file with constants for all of the schema types and the constants are all-uppercase letters (obviously that is usually a good convention for constants).  However, I am trying to generate a client api for a schema that I do not manage, but the schema has two types with the same letters, just different case (i.e. "Abc" and "ABC").  This causes the generated `DgsConstants` file to include 2 constants named `ABC` which does not compile.  I can work around this by manually changing the schema types (i.e. adding a "2" after one of them); however, I'd like to be able to just download the schema periodically (whenever another team makes changes to the schema) and hopefully not have to make these manual changes each time.

Question(s):
Is there a reason that the constants have to be all uppercase (other than convention)? or could there be an option added to keep the casing consistent with the type names?  For example, I see an option was added to allow for "snakeCaseConstantNames" so could an option be added to "matchCaseConstantTypeNames"?  Or is there already a config option that I am missing to tweak this behavior?

**Comment by srinivasankavitha** [comment, marked as answer]
Hi @thedamones - the reason we have uppercase  is purely convention. We don't yet have a config option to control the behavior, but should be fairly easy to add one. Feel free to open a PR if this is something you need sooner than later. Else we can create a ticket and prioritize on our end, but that might be further out in the quarter.

**Reply by devdamone** [reply]
Thank you for the response.  I played around with adding a config option in just the same way that "snakeCaseConstantNames" was added and that did seem easy and does work for my use-case.  However, I want to play around with it some more just to make sure I understand the rest of plugin and testing before opening a PR.  So for now I will keep with doing the manual changes, but hopefully I can find some time to open a PR for this soon.
